### ISSUE TITLE
Speak typed characters and words applied to edit fields only.

### ISSUE THREAD

**Issue by SamuelV04** [nvaccess/nvda#16848]
### Is your feature request related to a problem? Please describe.
No.
### Describe the solution you'd like
NVDA should be able to speak characters in edit fields, but act like speak typed characters is off in other places. For example, NVDA could continue to speak characters in notepad, but in file explorer, NVDA will not speak the characters, even if speak typed characters is turned on. If the user wishes, they can select whether to have NVDA not speak characters, speak characters everywhere, or only in edit boxes. The INSERT+2 keystroke would be able to cycle between the 3, and the speak typed characters checkbox in keyboard settings would be replaced by a combo box to select which setting the user wants. This would go for speak typed words as well.
### Describe alternatives you've considered
There is the typing settings addon that allows for this, but it isn't compatible with the latest NVDA.
### Additional context
More often than not, we are not working in an edit field. In areas such as file explorer, It can be somewhat verbose when using first letter navigation because the characters are being echoed every time they move to a new item in say, a list or a combobox. The user would end up turning speak typed characters off, and potentially forget to turn it back on when inside of an edit field. My suggestion would eliminate that problem, allowing the user greater flexibility in how they want things echoed.

**Comment by XLTechie** [comment]
@Adriani90 I am not convinced that this is a duplicate of #10331.

While I agree that #10331 is more important because of the typed words implications, the proposal here to include an option for typed characters to include an "off, everywhere, edit fields only" configuration, is not duplicated there.

**Comment by SamuelV04** [comment]
Right. #10331 doesn't have anything whatsoever about configuring how typed characters and words are echoed. Also, what if the user wants characters and words echoed everywhere as in prior versions? Like new keyboard learners for instance. There isn't any configurability mentioned in the issue, which to me makes it not a duplicate. People are going to have situations where they will want characters spoken globally in one application, while keeping them on only in edit boxes or off entirely in others.

**Comment by XLTechie** [comment]
Re-opening per https://github.com/nvaccess/nvda/issues/16848#issuecomment-2223718866
I don't necessarily support the feature request, but it is a valid request and IMO not a duplicate.

**Comment by cary-rowen** [comment]
Hi,

I developed an add-on some time ago with features that I thought could cover this feature request.
And I'd like to contribute this as a PR to NVDA core. It may involve configuration key changes, which I think would be best done in 2025.1.

Add-on repo:
https://github.com/cary-rowen/NVDA-CustomSpeakTypedCharacters

**Comment by seanbudd** [comment]
@cary-rowen this issue is triaged - is there anything you are waiting on from our side?

**Comment by cary-rowen** [comment]
Hi @seanbudd 
This involves a configuration key change that wouldn't require setting up backwards compatibility if done during the 2025.1 development cycle, right?

Many thanks!

**Comment by seanbudd** [comment]
Correct, if it comes after 2025.1 we will need to figure out a compatibility strategy
